### PR TITLE
View selector

### PR DESCRIPTION
**Pull request by terezanovotna**
## Description
* Adding View Selector Pattern

#575

### CODE REVIEW

File: pattern-library/forms-and-controls/view-selector/design/overview.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# View Selector
+
+The view selector is located in a toolbar providing views for a [List View](https://www.patternfly.org/pattern-library/content-views/list-view/), [Card View](http://www.patternfly.org/pattern-library/content-views/card-view/) or [Table View](http://www.patternfly.org/pattern-library/content-views/table-view/).
+
+View selector icons should be placed in this order (1) List View (2) Card View (3)Table View - optional. These controls may be used either together or individually.
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/view-selector/design/overview.md]: Not exactly sure what you mean by the last sentence.

Comment (mcarrano) [inline pattern-library/forms-and-controls/view-selector/design/overview.md]: I'm thinking this belongs on the Design page.

File: pattern-library/forms-and-controls/view-selector/design/design.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# View Selector
```
Comment (mcarrano) [inline pattern-library/forms-and-controls/view-selector/design/design.md]: We probably want a mockup that shows the placement of these icons in a toolbar. Are there any situations where a View Selector might be used independent of a content view @serenamarie125 ?

File: pattern-library/forms-and-controls/view-selector/design/overview.md (hunk 1 above)
Comment (terezanovotna) [reply pattern-library/forms-and-controls/view-selector/design/overview.md]: Will delete last sentence.👍

File: pattern-library/forms-and-controls/view-selector/design/design.md (hunk 2 above)
Comment (terezanovotna) [reply pattern-library/forms-and-controls/view-selector/design/design.md]: There's a PR with icon placement [showed on a toolbar](https://github.com/patternfly/patternfly-design/pull/621/commits/a8f8206c49f7c0010ef32d3a528a436b7440e6b4). Is that sufficient? Or do you want an example here in View Selector as well?

File: pattern-library/forms-and-controls/view-selector/design/overview.md (hunk 1 above)
Comment (terezanovotna) [reply pattern-library/forms-and-controls/view-selector/design/overview.md]: I have no problem with moving the image View Selector Detail to design page, but I think leaving it here brings more context. Leave it or move it?

File: pattern-library/forms-and-controls/view-selector/design/design.md (hunk 2 above)
Comment (mcarrano) [reply pattern-library/forms-and-controls/view-selector/design/design.md]: Now that I look at this again, I'm wondering if we should just add this to the Toolbar pattern and not create a new View Selector pattern. There does not seem like enough to say about this on it's own. What do you think @serenamarie125 ?

Comment (serenamarie125) [reply pattern-library/forms-and-controls/view-selector/design/design.md]: I think the reason we initially asked for this to be a separate pattern was due to the fact that we are trying to do the same for actions.  View selector could be utilized in a variety of locations, similar to actions.

Hypothetically they could be utilized on the same bar as a breadcrumb OR shared on a toolbar with filters.

Comment (terezanovotna) [reply pattern-library/forms-and-controls/view-selector/design/design.md]: thanks @serenamarie125. So if we keep it as a separate pattern, Matt - how does this PR look like? did I miss anything?

**Comment by mcarrano** [comment]
@terezanovotna So if we are to keep this as a separate pattern, than I think we need to include a bit more information. 

On the Overview page...
* Could we also include a mockup that shows what this would look like outside of the context of a toolbar? @serenamarie125 do we have any examples of that usage? Similar to what we are doing for actions?
* We should also include some statement of what the View Selector actually does, when you would use one and why.

On the Design page...
Showing a mockup with callouts will be more consistent with the way we've done other patterns. See http://www.patternfly.org/pattern-library/forms-and-controls/sort/#design for an example of another similar toolbar widget.

**Comment by terezanovotna** [comment]
I implemented the changes we agreed on in the last call. Please let me know what you think! https://docs.google.com/document/d/19YRgDY1FeSm72zYDyNg4E5daWyjMAh5XJGBb_SSUpiM/edit?usp=sharing @mcarrano @serenamarie125

**Pull request merged**